### PR TITLE
WIP: Initial glossary from workshop.

### PR DESCRIPTION
**Pull request by happi**
WIP
Please review and comment.

If you have some excellent diagrams to make things clearer, feel free to add them.

When satisfied, we should copy this glossary to the aeternity repo and check the code to ensure we use the correct terminology.

### CODE REVIEW

File: glossary.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# Glossary
+### CC Block
+Contains one keyblock and possibly one microblock
+
+### Keyblock
+A placeholder block indicating the leader that should produce microblocks in this generation.
```
Comment (ThomasArts) [inline glossary.md]: "generation" and "plural micro blocks is confusing here.
```suggestion
A placeholder block indicating the leader that should produce a microblock.
```

File: glossary.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+### Microblock
+A block with transactions.
+
+### Epoch
+Period of time represented in blocks during which validators are pre-selected to produce blocks and after which the state of the CC can be pinned to the PC.
```
Comment (ThomasArts) [inline glossary.md:30]: Possibly call this a "CC Epoch then".
There's also epoch for PC, but that differs in definition

File: glossary.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+Tokens minted and/or transferred to staking contract to reward successful pinning operation.
+
+### Leader Election
+The weighted random selection of leader from among eligible validators based on the amount to tokens staked in a staking contract.
+
+### Leader Election Contract
+Election rules defined at contract level (as opposed to consensus level). Likely not part of v1.0.
+
+### Genesis Accounts
+Accounts in which HC tokens are generated during the first block of CC.
```
Comment (ThomasArts) [inline glossary.md:143]: In our case it also contains some contracts used in the CC... or actually, no, but we have the notion of genesis contracts or genesis block as well

Comment (happi) [reply glossary.md:143]: Could you formulate that as a change request?

**Comment by happi** [comment]
The diagram looks nice in my VSCode, but it is a bit messed up on GitHub. Do you get the right view, or should I add a rendered picture?
image

File: glossary.md [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+The act of creation of a block done by the producer
+
+
+### Block Reward
+Tokens minted and/or transferred to staking contract to reward successful block production.
+
+### Pinning Reward
+Tokens minted and/or transferred to staking contract to reward successful pinning operation.
+### Halting
+A situation when no more blocks are produced (due to all stakers being inactive). This creates a problem for the HC creator to recover the HC and resume the block production. Some possible solutions are proposed.
```
Comment (dincho) [inline glossary.md]: Isn't stakers defined as passive actors above? Shouldn't this refer to validators?

> due to all validators being inactive

File: glossary.md [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+Willful end of a hyperchain
+
+## On-Chain Protocol Components
+
+### Staking Contract
+Contract used by validator to represent the validator's stake, the ID and stake amount of delegators, and contains the functions for withdrawal of rewards.
+
+
+### Leader Election Contract
+Election rules defined at contract level (as opposed to consensus level). Likely not part of v1.0.
```
Comment (hanssv) [inline glossary.md]: Given that the current broken implementation has (parts of) its consensus in a contract it is safe to assume that whatever we build also will be a hybrid-solution.

File: glossary.md [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+### Parent Chain (PC)
+Any PoW blockchain that fulfills the role of "the other half" of a hyperchain by providing entropy for the CC and allowing for the pinning of data from the CC. The parent chain functions independently from and has no knowledge of the child chain.
+
+## Blocks
+
+### CC Block
+Contains one keyblock and possibly one microblock.
+
+### Keyblock
+A placeholder block indicating the leader that should produce a microblock.
```
Comment (hanssv) [inline glossary.md]: Not sure about the placeholder comment. The keyblock contains all the state tree hashes etc so it is crucial for consensus. However, the separation into micro-block + key-block is entirely artificial and a legacy of building on top of a code base that uses BitcoinNG...

**Pull request merged**